code/game/objects/items/devices/reportintercom.dm

```diff
@@ -0,0 +1,70 @@
+/obj/item/device/reportintercom
+	icon = 'icons/obj/radio.dmi'
+	name = "hacked intercom"
+	desc = "Looks like a typical intercom. Except for the wires conspiciously hanging out. 'PROPERTY OF NANOTRASEN CENTRAL COMMAND' is stamped on the side."
+	icon_state = "intercom-open"
+	item_state = "intercom-open"
+	w_class = W_CLASS_TINY
+	flags = FPRINT
+	siemens_coefficient = 1
+
+	var/used = 0
+
+/obj/item/device/reportintercom/attack_self(mob/user) // can create a custom announcement once
+	if(used)
+		to_chat(user, "<span class='notice'>You press the [src]'s button, but nothing happens.</span>")
+		return
+
+	if(!map.linked_to_centcomm)
+		var/confirmation = alert("The station is not linked to central command by a relay. Blow your cover?",,"Yes","No")
+		if(confirmation == "No")
+			return
+	var/input = input(user, "Please enter anything you want. Anything.", "What?", "") as message|null
+	var/customname = input(user, "Pick a title for the report.", "Title") as text|null
+	if(!input)
+		return
+	if(!customname)
+		customname = "Nanotrasen Update"
+	for (var/obj/machinery/computer/communications/C in machines)
+		if(! (C.stat & (BROKEN|NOPOWER) ) )
+			var/obj/item/weapon/paper/P = new /obj/item/weapon/paper( C.loc )
+			P.name = "'[command_name()] Update.'"
+			P.info = input
+			P.update_icon()
+			C.messagetitle.Add("[command_name()] Update")
+			C.messagetext.Add(P.info)
+
+	switch(alert("Should this be announced to the general population?",,"Yes","No"))
+		if("Yes")
+			command_alert(input, customname,1);
+		if("No")
+			to_chat(world, "<span class='warning'>New Nanotrasen Update available at all communication consoles.</span>")
+
+	world << sound('sound/AI/commandreport.ogg', volume = 60)
+	log_admin("[key_name(user)] has created a fake command report: [input]")
+	message_admins("[key_name_admin(user)] has created a fake command report", 1)
+	feedback_add_details("admin_verb","FAKECCR")
+
+	icon_state = "intercom-p-open"
+	desc = "Looks like a typical intercom. The wires look burnt out. 'PROPERTY OF NANOTRASEN CENTRAL COMMAND' is stamped on the side."
+	used = 1
+	return 1
+
+/obj/item/device/reportintercom/attackby(obj/item/I, mob/user) //Please don't do this
```

Comment:
Would be better in `emag_act`

Reply:
https://github.com/vgstation-coders/vgstation13/blob/08f4abdf6e759c1a0663aa67d54eae63fb27d661/code/game/objects/items/devices/reportintercom.dm#L53-L57

should be changed to emag_act and still work now (tested with emag)